src/components/shared/Pagination/Pagination.test.js

```diff
@@ -0,0 +1,142 @@
+import React from 'react';
+import { render, cleanup } from '@testing-library/react';
+import '@testing-library/jest-dom/extend-expect';
+import { Pagination } from './Pagination';
+import { BrowserRouter } from 'react-router-dom';
+
+describe('Pagination Component', () => {
+  afterEach(cleanup);
+
+  it('should show previous page element when current page is not the first one', () => {
+    //Act
+    const { container } = render(
+      <BrowserRouter>
+        <Pagination currentPage={2} pagesCount={100} urlToGo={'https://domain.com/section?'} />
+      </BrowserRouter>,
+    );
+
+    // Asserts
+    expect(container.querySelector('.icon-arrow-prev')).toBeInTheDocument();
+  });
+
+  it('should show next page element when current page is not the last one', () => {
+    //Act
+    const { container } = render(
+      <BrowserRouter>
+        <Pagination currentPage={99} pagesCount={100} urlToGo={'https://domain.com/section?'} />
+      </BrowserRouter>,
+    );
+
+    // Asserts
+    expect(container.querySelector('.icon-arrow-next')).toBeInTheDocument();
+  });
+
+  it('should not show previous page element when current page is the first one', () => {
+    //Act
+    const { container } = render(
+      <BrowserRouter>
+        <Pagination currentPage={1} pagesCount={100} urlToGo={'https://domain.com/section?'} />
+      </BrowserRouter>,
+    );
+
+    // Asserts
+    expect(container.querySelector('.icon-arrow-prev')).not.toBeInTheDocument();
+  });
+
+  it('should not show next page element when current page is the last one', () => {
+    //Act
+    const { container } = render(
+      <BrowserRouter>
+        <Pagination currentPage={100} pagesCount={100} urlToGo={'https://domain.com/section?'} />
+      </BrowserRouter>,
+    );
+
+    // Asserts
+    expect(container.querySelector('.icon-arrow-next')).not.toBeInTheDocument();
+  });
+
+  it('should show ... to slide to left when selected page is not the first one', () => {
```

Comment:
Here shouldn´t it be shown the points when you are farther away from the begining? like at least 5 pages away?

Reply:
It could be an improvement.
If you are in page 3 you will see:
< 1 ... 3 4 5 6 7 ... 100>

if you press the first ... you will see:
< 1 ... 2 3 4 5 6 ... 100 >

if you press it again you will see:
< 1 2 3 4 5 ... 100 >

The idea of the ... is allow you to move between pages without going to any one until you select one.

Reply:
Thanks for the explanation.